dd-sdk-android/src/main/kotlin/com/datadog/android/rum/gestures/DatadogGesturesListener.kt

```diff
@@ -0,0 +1,152 @@
+        decorViewReference.get()?.let {
+            // we just intercept but not steal the event
+            val spanBuilder = rumTracer
+                .buildSpan(UI_TAP_ACTION_EVENT)
+            val target =
+                findTarget(
+                    it,
+                    e.x,
+                    e.y
+                )
+            spanBuilder.withTag(TAG_TARGET_CLASS_NAME, target.javaClass.canonicalName)
+            // we will handle general exceptions here. We don't know what might go wrong apart from
+            // ResourceNotFound exception.
+            @Suppress("TooGenericExceptionCaught")
+            try {
+                spanBuilder.withTag(
+                    TAG_TARGET_RESOURCE_ID,
+                    target.resources.getResourceEntryName(target.id)
+                )
+            } catch (e: Exception) {
+                sdkLogger.e("$TAG: Could not found resource name for target:${target.id}", e)
+            }
+            spanBuilder
+                .start()
+                .finish(DEFAULT_EVENT_DURATION)
+        }
+        return false
+    }
+
+    override fun onDown(e: MotionEvent): Boolean {
+        // No Op
+        // we just intercept but not steal the event
+        return false
+    }
+
+    override fun onFling(
+        e1: MotionEvent,
+        e2: MotionEvent,
+        velocityX: Float,
+        velocityY: Float
+    ): Boolean {
+        // No Op
+        // we just intercept but not steal the event
+        return false
+    }
+
+    override fun onScroll(
+        e1: MotionEvent,
+        e2: MotionEvent,
+        distanceX: Float,
+        distanceY: Float
+    ): Boolean {
+        // No Op
+        // we just intercept but not steal the event
+        return false
+    }
+
+    override fun onLongPress(e: MotionEvent) {
+        // No Op
+    }
+
+    // endregion
+
+    companion object {
+        private const val TAG = "DatadogGestureListener"
+        internal const val UI_TAP_ACTION_EVENT = "TapEvent"
+        internal const val TAG_TARGET_CLASS_NAME = "target.classname"
+        internal const val TAG_TARGET_RESOURCE_ID = "target.resourceId"
+        val DEFAULT_EVENT_DURATION = TimeUnit.MILLISECONDS.toMicros(16)
+
+        private fun findTarget(decorView: View, x: Float, y: Float): View {
+            val stack = LinkedList<View>()
+            stack.addFirst(decorView)
+            while (stack.isNotEmpty()) {
+                val view = stack.removeFirst()
+
+                if (isValidTarget(view)) {
+                    return view
+                }
+
+                if (view is ViewGroup) {
+                    handleViewGroup(view, x, y, stack)
+                }
+            }
+
+            return decorView
+        }
+
+        private fun handleViewGroup(
+            view: ViewGroup,
+            x: Float,
+            y: Float,
+            stack: LinkedList<View>
+        ) {
+            for (i in 0 until view.childCount) {
```

Comment:
Shouldn't we also consider z-order of the views? The frontmost views should have priority in this iteration.

Reply:
hmmm...I am not sure I want to add this for now ....it will add more complexity here to sort them but I will have this in mind if this doesn't work as it should. I guess we need to wait and see how it behaves.